src/main/java/org/opensearch/performanceanalyzer/reader_writer_shared/EventLogFileHandler.java

```diff
@@ -166,4 +169,30 @@ private void readInternal(Path pathToFile, int bufferSize, EventDispatcher proce
             LOG.error("Error reading file", ex);
         }
     }
+
+    public void deleteFiles(long referenceTime, int purgeInterval) {
+        LOG.debug("Starting to delete old writer files");
+        long startTime = System.currentTimeMillis();
+
+        File root = new File(metricsLocation);
+        String[] children = root.list();
```

Comment:
Because the `EventLogQueueProcessor` generates the files and also provides the arguments `referenceTime` and `purgeInterval`, I suggest it keep track of all the files that are to be deleted and just provide that as an input. That way we can avoid a directory scan.

Reply:
Done. 
We will now scan only once, for the first purge after the start-up to address any lingering event log files. 
All other executions, we will only lookup and delete the past `filesCleanupPeriod` files.